opacus/grad_sample/grad_sample_module.py

```diff
@@ -125,6 +124,7 @@ def __init__(
 
         for p in self.parameters():
```

Comment:
<Comment unrelated to this specific PR/issue.>
This had not caught my eye before. Is there any reason why we attach `.grad_sample` to all parameters instead of just the trainable ones?

Reply:
Weeell, I don't think there's a reason. 
I always thought we've had a util method that does '[p for p in model.parameters() if p.requires_grad]', but apparently no, and we're consistently using just `model.parameters()` all over